### PR TITLE
Add new export experience

### PR DESCRIPTION
**Pull request by rijkvanzanten**
Upgrades the exporting experience by allowing configuration of the to-be exported set in a drawer:

CleanShot 2022-03-16 at 18 28 54@2x

and by allowing exporting of data sets of any length by adding an option to save exports directly to the file library. 

Left to-do:

- [x] Add docs (env var, API endpoint, app usage)
- [x] Update full width notice to have helpful information on the async nature of save-to-file library
- [x] Add max-item limit for direct downloads
- [x] Fix cancel button in drawer

### CODE REVIEW

**Comment by nodeworks** [comment]
@rijkvanzanten will this allow for adding relational fields and their display values instead of their ID's?

File: app/src/interfaces/_system/system-fields/system-fields.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+					<div class="spacer" />
+					<v-icon name="close" clickable @click="removeField(field.key)" />
+				</v-list-item>
+			</template>
+		</draggable>
+	</v-list>
+	<v-menu placement="bottom-start" show-arrow>
+		<template #activator="{ toggle }">
+			<button class="toggle" @click="toggle">
+				Add Field
```
Comment (rijkvanzanten) [inline app/src/interfaces/_system/system-fields/system-fields.vue:17]: ⚠️

File: app/src/modules/content/routes/collection.vue [hunk 2]
```diff
@@ -248,6 +248,7 @@
 					:collection="collection"
 					:filter="mergeFilters(filter, archiveFilter)"
 					:search="search"
+					:layout-query="layoutQuery"
```
Comment (rijkvanzanten) [inline app/src/modules/content/routes/collection.vue:251]: ⚠️ add to other modules as well

File: app/src/views/private/components/export-sidebar-detail.vue [hunk 3]
```diff
@@ -51,221 +51,455 @@
...
+				<v-button small full-width @click="exportDialogActive = true">
+					{{ t('export_items') }}
+				</v-button>
 			</div>
+		</div>
 
-			<div class="field full">
-				<v-button small full-width @click="exportData">
-					{{ t('export_data_button') }}
+		<v-drawer
```
Comment (rijkvanzanten) [inline app/src/views/private/components/export-sidebar-detail.vue:60]: ⚠️ Add correct icon

File: app/src/views/private/components/export-sidebar-detail.vue [hunk 4]
```diff
@@ -51,221 +51,455 @@
...
 </template>
 
-<script lang="ts">
-import { useI18n } from 'vue-i18n';
-import { defineComponent, ref, PropType, computed } from 'vue';
+<script lang="ts" setup>
+import api from '../../../api';
+import { getRootPath } from '../../../utils/get-root-path';
+import { notify } from '../../../utils/notify';
+import readableMimeType from '../../../utils/readable-mime-type';
```
Comment (rijkvanzanten) [inline app/src/views/private/components/export-sidebar-detail.vue:215]: ⚠️ Use `@/` instead, and figure out why Nova 3 is being difficult with aliases

**Comment by nodeworks** [comment]
@rijkvanzanten 

One last question on export, if you export a M2M relationship in CSV format it displays the JSON in the cell instead of the actual value. Is it possible to get just the value?

For example it displays as:

`[{"regions_id":{"short_name":"SOMEREGION"}}, {"regions_id":{"short_name":"SOMEOTHERREGION"}}]`

Would it be feasible to have it be:

`"SOMEREGION,SOMEOTHERREGION"`

**Comment by rijkvanzanten** [comment]
Ohh good one 🤔 Mind opening a new discussion for that so it doesn't get lost in the sauce here?

**Comment by nodeworks** [comment]
> Ohh good one 🤔 Mind opening a new discussion for that so it doesn't get lost in the sauce here?

No problem:
https://github.com/directus/directus/discussions/12288

**Pull request merged**